### PR TITLE
feat: created backend for virtual assessment

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes
-  Created few endpoints to create/delete/get virtual assessment
-  Created new table virtual assessments
Screenshot from 2023-12-07 20-44-39
Screenshot from 2023-12-07 21-37-12

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://homecareaid.atlassian.net/jira/software/projects/CC/boards/1/backlog?issueParent=10117&selectedIssue=CC-124)

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/virtual_assessment/virtual_assessment.module.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Module({
+  imports: [
+    TypeOrmModule.forFeature([VirtualAssessment, Appointment]),
+    JwtModule.registerAsync({
+      useFactory: async (configService: ConfigService) => ({
+        secret: configService.get('JWT_SECRET'),
+        signOptions: { expiresIn: configService.get('JWT_TOKEN_EXPIRE') },
+      }),
+      inject: [ConfigService],
+    }),
```
Comment (Simpsonkk) [inline src/modules/virtual_assessment/virtual_assessment.module.ts]: our JwtModule is global therefore you can use it in virtual assessment without importing into the module.

Comment (ProggerOleg) [reply src/modules/virtual_assessment/virtual_assessment.module.ts]: You're right, fixed

File: src/modules/virtual_assessment/virtual_assessment.service.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        .leftJoinAndSelect('virtualAssessment.appointment', 'appointment')
+        .where('appointment.id = :id', { id: appointmentId })
+        .getOne();
+
+      if (!virtualAssessment) {
+        throw new HttpException(
+          ErrorMessage.VirtualAssessmentNotFound,
+          HttpStatus.NOT_FOUND,
+        );
+      }
```
Comment (Simpsonkk) [inline src/modules/virtual_assessment/virtual_assessment.service.ts]: You can use your findVirtualAssessmentById to avoid duplication.

**Comment by Simpsonkk** [comment]
image

This route looks like you wanna delete the appointment but the description says that you wanna delete virtual assessment

**Comment by ProggerOleg** [comment]
Wrong screenshot was uploaded, fixed)

File: src/modules/virtual_assessment/virtual_assessment.controller.ts
Comment (Simpsonkk) [inline src/modules/virtual_assessment/virtual_assessment.controller.ts:1]: we use "-" for multi-word naming, not "_"

File: src/modules/virtual_assessment/virtual_assessment.service.ts (hunk 2 above)
Comment (ProggerOleg) [reply src/modules/virtual_assessment/virtual_assessment.service.ts]: Agree, fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**